activiti-core/activiti-api-impl/activiti-api-process-runtime-impl/src/test/java/org/activiti/runtime/api/impl/ProcessRuntimeImplTest.java

```diff
@@ -175,4 +187,56 @@ public void should_throwActivitiObjectNotFoundException_when_canReadFalse() {
             .hasMessage("Unable to find process definition for the given id:'processDefinitionId'");
     }
 
+    @Test
+    public void should_createAprocessInstance_whenCreateIsCalled() {
```

Comment:
IMO, this test is two complex because it became a mix of unit test and integration test (I know, other tests in the class are doing the same, but they shouldn't :)). I think that at this level `RepositoryService` should be a mock, then there should be no reference to `CommandExecutor`. You should simply check that `ValidatorService` `RuntimeService` and  `builder` are called.

Then, in addition to this unit test, you should have an Integration Test in the starter covering the integration aspects. Could be added [here](https://github.com/Activiti/Activiti/blob/develop/activiti-core/activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/process/ProcessRuntimeIT.java).